packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -304,83 +307,66 @@ describe('ComboBox', function () {
   });
 
   it('can be disabled', async function () {
-    let {getByRole, queryByRole} = renderComboBox({isDisabled: true});
+    let tree = renderComboBox({isDisabled: true});
+    let {setInteractionType, setElement, getCombobox, getListbox, open} = testUtilUser.createTester('ComboBoxTester');
+    setElement(tree.container);
 
-    let combobox = getByRole('combobox');
     act(() => {
-      combobox.focus();
+      getCombobox().focus();
     });
     await user.keyboard('One');
-    act(() => {
-      jest.runAllTimers();
-    });
-
-    expect(queryByRole('listbox')).toBeNull();
+    expect(getListbox()).toBeFalsy();
     expect(onOpenChange).not.toHaveBeenCalled();
     expect(onFocus).not.toHaveBeenCalled();
 
-    await user.keyboard('{ArrowDown}');
-    act(() => {
-      jest.runAllTimers();
-    });
+    setInteractionType('keyboard');
+    await open();
 
-    expect(queryByRole('listbox')).toBeNull();
+    expect(getListbox()).toBeFalsy();
     expect(onOpenChange).not.toHaveBeenCalled();
 
-    let button = getByRole('button');
-    await user.click(button);
-    act(() => {
-      jest.runAllTimers();
-    });
-
-    expect(queryByRole('listbox')).toBeNull();
+    setInteractionType('mouse');
+    await open();
+    expect(getListbox()).toBeFalsy();
     expect(onOpenChange).not.toHaveBeenCalled();
     expect(onInputChange).not.toHaveBeenCalled();
   });
 
   it('can be readonly', async function () {
-    let {getByRole, queryByRole} = renderComboBox({isReadOnly: true, defaultInputValue: 'Blargh'});
+    let tree = renderComboBox({isReadOnly: true, defaultInputValue: 'Blargh'});
+    let {setInteractionType, setElement, getCombobox, getListbox, open} = testUtilUser.createTester('ComboBoxTester');
+    setElement(tree.container);
 
-    let combobox = getByRole('combobox');
     act(() => {
-      combobox.focus();
+      getCombobox().focus();
     });
     await user.keyboard('One');
-    act(() => {
-      jest.runAllTimers();
-    });
-
-    expect(queryByRole('listbox')).toBeNull();
-    expect(combobox.value).toBe('Blargh');
+    expect(getListbox()).toBeFalsy();
+    expect(getCombobox().value).toBe('Blargh');
     expect(onOpenChange).not.toHaveBeenCalled();
     expect(onFocus).toHaveBeenCalled();
     expect(onInputChange).not.toHaveBeenCalled();
 
-    await user.keyboard('{ArrowDown}');
-    act(() => {
-      jest.runAllTimers();
-    });
+    setInteractionType('keyboard');
+    await open();
```

Comment:
I opted for `setInteractionType` instead because it felt more annoying to continually set the `interactionType` every time an interaction was called. I figured users would more often than not stick with a single interaction type for most of their interactions.

Reply:
Maybe we have a default + setter + local option override? then you don't have to change it and change it back if you just want a one off interaction?

Reply:
so support `await open({interactionType: 'keyboard'});` in addition to `setInteractionType` where the local option wins if provided? Sounds good to me, will add

Reply:
Updated to support this in all the existing utils